rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3382,6 +3384,226 @@ int LuaUnsyncedCtrl::SetBuildFacing(lua_State* L)
 }
 
 
+static bool ParseBuildInfo(
+	lua_State* L,
+	BuildInfo& buildInfo,
+	const char* caller,
+	int idx
+) {
+	for (int i=0; i<5; i++) {
+		if (!lua_isnumber(L, idx+i)) {
+			luaL_error(L, "%s(): bad build info type", caller);
+			return false;
+		}
+	}
+
+	int unitDefID = lua_toint(L, idx);
+	float3 pos = {lua_tofloat(L, idx+1), lua_tofloat(L, idx+2), lua_tofloat(L, idx+3)};
+	int facing = lua_toint(L, idx+4);
+
+	buildInfo = BuildInfo(unitDefID, pos, facing);
+	return true;
+}
+
+
+static bool ParseBuildInfoTable(
+	lua_State* L,
+	BuildInfo& buildInfo,
+	const char* caller,
+	int idx
+) {
+	if (!lua_istable(L, idx)) {
+		luaL_error(L, "%s(): bad build info type", caller);
+		return false;
+	}
+
+	lua_rawgeti(L, idx, 1);
+	if (!lua_isnumber(L, -1)) {
+		luaL_error(L, "%s(): bad unit def ID", caller);
+		return false;
+	}
+	int unitDefID = lua_toint(L, -1);
+	lua_pop(L, 1);
+
+	float3 pos;
+	for (int i=0; i<3; i++) {
+		lua_rawgeti(L, idx, i+2);
+		if (!lua_isnumber(L, -1)) {
+			luaL_error(L, "%s(): bad pos", caller);
+			return false;
+		}
+		pos[i] = lua_tofloat(L, -1);
+		lua_pop(L, 1);
+	}
+
+	lua_rawgeti(L, idx, 5);
+	if (!lua_isnumber(L, -1)) {
+		luaL_error(L, "%s(): bad facing", caller);
+		return false;
+	}
+	int facing = lua_toint(L, -1);
+	lua_pop(L, 1);
+
+	buildInfo = BuildInfo(unitDefID, pos, facing);
+	return true;
+}
+
+
+static bool ParseBuildInfoArray(
+	lua_State* L,
+	std::vector<BuildInfo>& buildInfos,
+	const char* caller,
+	int idx
+) {
+	if (!lua_istable(L, idx)) {
+		luaL_error(L, "%s(): error parsing build info array", caller);
+		return false;
+	}
+
+	for (lua_pushnil(L); lua_next(L, idx) != 0; lua_pop(L, 1)) {
+		buildInfos.emplace_back();
+		if (!ParseBuildInfoTable(L, buildInfos.back(), caller, -1)) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+
+static bool ParseBuildSquareOptions(
```

Comment:
I'd remove the explicit color arguments now that cmdColors for build squares exist.

Reply:
done